.travis.yml

```diff
@@ -20,114 +20,123 @@
+     - ipython
+     - cython
+     - pkg-config
+     - openmpi-bin
+     - libopenmpi-dev
+     - python-nose
+     - libpcre3
+     - libpcre3-dev
+     - python-mpi4py
+     - llvm-3.6-dev
+     - jq
+     - cppcheck
+     - vera++
+     - pep8
 cache:
-  directory:
-    - $HOME/.cache
+   directory:
+     - $HOME/.cache
 
 before_install:
-  - echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
-  - echo "+         P R E P A R E   N E S T   B U I L D   E N V I R O N M E N T         +"
-  - echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
-  - export SOURCEDIR=$PWD
-  - cd $HOME/build
-  - echo $PATH
-
-  # Update package repository status (-qq is more quiet).
-  - sudo rm -rf /var/lib/apt/lists/*
-  - ls /etc/apt/sources.list.d/
-  - sudo apt-get update -qq
-
-  # Remove automatically all unused packages.
-  - sudo apt-get autoremove
-
+   - echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
+   - echo "+         P R E P A R E   N E S T   B U I L D   E N V I R O N M E N T         +"
+   - echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
+   - export SOURCEDIR=$PWD
+   - echo $PWD
+   - cp music_install.sh $HOME
+   - cd $HOME/build
+   - echo $PATH
+  # Upgrade pip and setuptools
+   - pip install -U pip
+   - pip install -U setuptools
+  
   # Install packages required by NEST.
   # (Please see also: http://www.nest-simulator.org/installation/#standard-configuration)
-  - sudo apt-get install -y build-essential cmake libltdl7-dev libreadline6-dev libncurses5-dev libgsl0-dev python-all-dev python-numpy python-scipy python-matplotlib ipython
-
-  # Install pip and Cython.
-  - wget https://bootstrap.pypa.io/get-pip.py
-  - sudo python get-pip.py
-  - sudo pip install -U setuptools
-  - pip install --user cython==0.23.4
-
-  # Install package terminaltables required by parse_travis_log.py to create the build summary.
-  - sudo pip install terminaltables
-
-  # Install OpenMPI.
-  - sudo apt-get install -y openmpi-bin libopenmpi-dev
-
-  # Install nose framework for the Python testsuite.
-  - sudo apt-get install -y python-nose
-
-  # Install VERA++ and pep8 for static code analysis.
-  - sudo apt-get install -y vera++ pep8
-
-  # Required for building cppcheck-1.69.
-  - sudo apt-get install -y libpcre3 libpcre3-dev
-
-  # Install jq to process JSON output from GitHub API.
-  - sudo apt-get install -y jq
-
-  # Install current MUSIC version.
-  - sudo apt-get install -y python-mpi4py
-  - git clone https://github.com/INCF/MUSIC.git music
-  - cd music
-  - ./autogen.sh
-  - ./configure --prefix=$HOME/.cache/music.install
-  - make
-  - make install
-  - cd $HOME/build
-
-  # Change directory back to the NEST source code directory.
-  - cd $SOURCEDIR
+  # terminaltables is required by parse_travis_log.py to create the build summary.
+   - pip install cython scipy matplotlib terminaltables mpi4py
```

Comment:
`scipy`, `matplotlib` and `mpi4py` are already installed above in the `apt` block. Is `pip install`ing these packages a requirement so that the test setup works with the different Python versions?

Can you please either remove the packages either here (my preferred solution if that works with testing different Python versions) or above, so they are not installed multiple times? Also, could `cython` and `terminaltables` be installed using `apt-get`?

Reply:
No more using the deprecated version. Since the version of some packages like scipy is old in the source package whitelist, I am installing those via pip